pkg/apis/acid.zalan.do/v1/postgresql_type.go

```diff
@@ -225,9 +226,48 @@ type Sidecar struct {
 // UserFlags defines flags (such as superuser, nologin) that could be assigned to individual users
 type UserFlags []string
 
+type Conditions []Condition
+
+type ConditionType string
+type VolatileTime struct {
+	Inner metav1.Time `json:",inline"`
+}
+
+// MarshalJSON implements the json.Marshaler interface.
+func (t VolatileTime) MarshalJSON() ([]byte, error) {
+	return t.Inner.MarshalJSON()
+}
+
+// UnmarshalJSON implements the json.Unmarshaller interface.
+func (t *VolatileTime) UnmarshalJSON(b []byte) error {
+	return t.Inner.UnmarshalJSON(b)
+}
+
+func init() {
+	equality.Semantic.AddFunc(
+		// Always treat VolatileTime fields as equivalent.
+		func(VolatileTime, VolatileTime) bool {
+			return true
+		},
+	)
+}
+
+// Condition contains the conditions of the PostgreSQL cluster
+type Condition struct {
+	Type               ConditionType      `json:"type" description:"type of status condition"`
+	Status             v1.ConditionStatus `json:"status" description:"status of the condition, one of True, False, Unknown"`
+	LastTransitionTime VolatileTime       `json:"lastTransitionTime,omitempty" description:"last time the condition transit from one status to another"`
```

Comment:
Mostly to avoid semantic check inequality with `metav1.time`, with a wrapper around `metav1.time` there won't be a semantic inequality even if the times are different but everything else in the CR is same, this could be handy while writing unit tests or e2e tests I think

Reply:
@FxKu , Is this okay?